src/Core/Domain/CartRule/QueryResult/CartRuleRestrictionsForEditing.php

```diff
@@ -42,18 +42,26 @@ class CartRuleRestrictionsForEditing
      */
     private array $productRestrictionRuleGroups;
 
+    /**
+     * @var int[]
+     */
+    private array $restrictedGroupIds;
+
     /**
      * @todo: fill other restrictions when related commands are implemented
      *
      * @param int[] $restrictedCartRuleIds
      * @param RestrictionRuleGroup[] $productRestrictionRuleGroups
+     * @param int[] $restrictedGroupIds
      */
     public function __construct(
         array $restrictedCartRuleIds,
-        array $productRestrictionRuleGroups
+        array $productRestrictionRuleGroups,
+        array $restrictedGroupIds
     ) {
         $this->restrictedCartRuleIds = $restrictedCartRuleIds;
         $this->productRestrictionRuleGroups = $productRestrictionRuleGroups;
+        $this->restrictedGroupIds = $restrictedGroupIds;
```

Comment:
I'm wondering if you can not use constructor property promotion here. 
It's for `develop` branch so it could be nice!
But I'm not block your PR for this.

Reply:
Hm indeed I will :+1: